src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -144,8 +146,43 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 			this._layoutCalledOnce = true;
 		}
 	}
+	reorderGridData(source: IDataResource) {
+		let rowIndex = 0;
+		// Get schema list of grid
+		const columnNames: string[] = source.schema.fields.map(field => field.name);
+		if (source.data.length > 0) {
+			// Check to see if data source is ordered properly based on schema
+			let rowKeys = Object.keys(source.data[0]);
+			if (!equals(columnNames, rowKeys)) {
+				// SQL notebooks does not use columnName as key (instead uses indices)
+				// Indicies indicate the row is ordered properly
+				// We check the data to know if it is in index form or string form
+				for (let index of rowKeys) {
+					// Check to see if first row keys are in index format (numbers)
+					let currentIndex = typeof Number(index);
```

Comment:
What if they have columns with a number as the name? You shouldn't need to cast here I would think - wouldn't the values for a SQL table be a number already instead of a string?

Reply:
SQL table row keys are in form of strings, and so in order to see if they are numbers I need to cast them to verify. There should not be numbers as row keys unless it is index format in the dataResource.

Reply:
Sigh, that's annoying.

Are you saying Kusto can't have numbers as column names? SQL can

`select 1 as [1], 2 as [2]`